tools/socktap/cam_application.cpp

```diff
@@ -76,8 +67,8 @@ void CamApplication::on_timer(const boost::system::error_code& ec)
     BasicContainer_t& basic = cam.camParameters.basicContainer;
     basic.stationType = StationType_passengerCar;
     basic.referencePosition.altitude.altitudeValue = AltitudeValue_unavailable;
-    basic.referencePosition.longitude = round(position.longitude, microdegree) * Longitude_oneMicrodegreeEast;
-    basic.referencePosition.latitude = round(position.latitude, microdegree) * Latitude_oneMicrodegreeNorth;
+    basic.referencePosition.longitude = std::round(static_cast<geo_angle_i32t>(position.longitude).value());
+    basic.referencePosition.latitude = std::round(static_cast<geo_angle_i32t>(position.latitude).value());
```

Comment:
Why do you prefer this approach?
I do not see the advantage at the moment and the previous code lines look more readable to me.

Reply:
I thought it would be better to re-use the existing conversion of the degree representation. However, I just realized that after casting to a `int32_t` quantity, the round should be a noop or am I missing something? Is it readability-wise okay when the round is omitted?

Apart from that, it's probably even better to move all code to extract/convert information from the `PositionFix` into a `CAM` to `cam_permissions.hpp/cpp`.

Reply:
Yes, calling `std::round` after casting to an integer type has no effect.
I am not a fan of calling `value()` on those `boost::units::quantity<>` because the result depends on the stored unit's scale. Just imagine what happens if GeoNetworking decides to use a different scale than the CDD… happy debugging then!
That's the reason why `microdegree` is passed in the present code: Making things explicit.

A helper function `to_reference_position(const PositionFix&): ReferencePosition_t` would be less verbose, agreed.